Test/com/nicoles_change_maker/test/ChangeMakerTest.java

```diff
@@ -0,0 +1,235 @@
+        change = changeMaker.makeChange(.05);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForSixCents(){
+        expectedChange.put("N", 1);
+        expectedChange.put("P", 1);
+        change = changeMaker.makeChange(.06);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForSevenCents(){
+        expectedChange.put("N", 1);
+        expectedChange.put("P", 2);
+        change = changeMaker.makeChange(.07);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForEightCents(){
+        expectedChange.put("N", 1);
+        expectedChange.put("P", 3);
+        change = changeMaker.makeChange(.08);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForNineCents(){
+        expectedChange.put("N", 1);
+        expectedChange.put("P", 4);
+        change = changeMaker.makeChange(.09);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForTenCents(){
+        expectedChange.put("D", 1);
+        change = changeMaker.makeChange(.10);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForElevenCents(){
+        expectedChange.put("D", 1);
+        expectedChange.put("P", 1);
+        change = changeMaker.makeChange(.11);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForTwelveCents(){
+        expectedChange.put("D", 1);
+        expectedChange.put("P", 2);
+        change = changeMaker.makeChange(.12);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForThirteenCents(){
+        expectedChange.put("D", 1);
+        expectedChange.put("P", 3);
+        change = changeMaker.makeChange(.13);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForFourteenCents(){
+        expectedChange.put("D", 1);
+        expectedChange.put("P", 4);
+        change = changeMaker.makeChange(.14);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForFifteenCents(){
+        expectedChange.put("D", 1);
+        expectedChange.put("N", 1);
+        change = changeMaker.makeChange(.15);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForSixteenCents(){
+        expectedChange.put("D", 1);
+        expectedChange.put("N", 1);
+        expectedChange.put("P", 1);
+        change = changeMaker.makeChange(.16);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForSeventeenCents(){
+        expectedChange.put("D", 1);
+        expectedChange.put("N", 1);
+        expectedChange.put("P", 2);
+        change = changeMaker.makeChange(.17);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForEighteenCents(){
```

Comment:
Also, this doesn't mean you shouldn't write these kind of tests along the way, especially if they're helpful stepping stones! Just that they might not always need to be permanent.